### PR TITLE
use SmallRng instead of ChaCha for generating test keys

### PR DESCRIPTION
**Pull request by eugene-babichenko**
Use a faster non-cryptographic generators for generating `Arbitrary` keys.

### CODE REVIEW

**Comment by eugene-babichenko** [comment]
`test_release` passes in less than 2 minutes now so this is an obvious improvement.

I wonder if we actually care about reproducible cryptographic keys? If not, we can just go with thread-local RNGs seeded from the system entropy, but this will require a significant refactoring. I tried to do that and there was no easy way to incorporate this change.

File: chain-crypto/src/testing.rs [hunk 1]
```diff
@@ -15,6 +15,36 @@ use rand_core::SeedableRng;
 #[derive(Clone, Debug)]
 pub struct TestCryptoGen(pub u64);
 
+pub struct TestCryptoRng(SmallRng);
```
Comment (mzabaluev) [inline chain-crypto/src/testing.rs:18]: Please document that this is a fake crypto RNG that does not really provide cryptograhic safety.

**Review by mzabaluev — APPROVED** [review]

**Comment by mzabaluev** [comment]
> I wonder if we actually care about reproducible cryptographic keys?

If a test fails, we'd like it to be reproducible by reusing the seed. This may be not how it works right now, but in principle all generators used in tests should be seedable. By default the seed should be initialized from entropy and printed. This may need to be done for every test case.

File: chain-crypto/src/testing.rs (hunk 1 above)
Comment (eugene-babichenko) [reply chain-crypto/src/testing.rs:21]: Done.

**Pull request merged**